packages/fuselage-ui-kit/src/elements/UsersSelectElement/UsersSelectElement.tsx

```diff
@@ -0,0 +1,64 @@
+import { AutoComplete, Box, Chip, Option } from '@rocket.chat/fuselage';
+import { useDebouncedValue } from '@rocket.chat/fuselage-hooks';
+import { UserAvatar } from '@rocket.chat/ui-avatar';
+import type * as UiKit from '@rocket.chat/ui-kit';
+import { useCallback, useMemo, useState } from 'react';
+
+import { useUiKitState } from '../../hooks/useUiKitState';
+import type { BlockProps } from '../../utils/BlockProps';
+import { useUsersData } from './hooks/useUsersData';
+
+type UsersSelectElementProps = BlockProps<UiKit.UsersSelectElement>;
+
+export type UserAutoCompleteOptionType = {
+  value: string;
+  label: string;
+};
+
+const UsersSelectElement = ({ block, context }: UsersSelectElementProps) => {
+  const [{ loading, value }, action] = useUiKitState(block, context);
+
+  const [filter, setFilter] = useState('');
+  const debouncedFilter = useDebouncedValue(filter, 1000);
+
+  const data = useUsersData({ filter: debouncedFilter });
+
+  const options = useMemo(() => data || [], [data]);
```

Comment:
You can remove this memo by returning `options || []` from the useQuery call itself.

Reply:
```suggestion
```